astropy/utils/data_info.py

```diff
@@ -231,46 +298,40 @@ class DataInfo:
 
     def __init__(self, bound=False):
         # If bound to a data object instance then create the dict of attributes
-        # which stores the info attribute values.
+        # which stores the info attribute values. Default of None for "unset"
+        # except for dtype where the default is object.
         if bound:
-            self._attrs = dict((attr, None) for attr in self.attr_names)
+            self._attrs = {}
 
     @property
     def _parent(self):
-        if self._parent_ref is None:
+        try:
+            return self._parent_ref()
+        except AttributeError:
             return None
-        else:
-            parent = self._parent_ref()
-            if parent is not None:
-                return parent
-
-            else:
-                raise AttributeError("""\
+        except TypeError:
```

Comment:
Very weird - there must have been an intermediate state where it didn't happen, since the test was failing. In any case, I'll fix and reinstate the test!